controllers/snapshot/snapshot_adapter.go

```diff
@@ -307,6 +308,27 @@ func (a *Adapter) EnsureCreationOfEphemeralEnvironments() (controller.OperationR
 					"integrationTestScenario.Name", integrationTestScenario.Name)
 				return controller.StopProcessing()
 			}
+			if clienterrors.IsInvalid(err) {
+
+				if !gitops.IsSnapshotMarkedAsFailed(a.snapshot) {
+					var (
+						errKind   string = "<unknown>"
+						errName   string = "<unknown>"
+						statusErr *clienterrors.StatusError
+					)
+					if ok := errors.As(err, &statusErr); ok {
+						errKind = statusErr.Status().Details.Kind
+						errName = statusErr.Status().Details.Name
+					}
+					a.logger.Error(err, fmt.Sprintf("Resource %s (%s) referenced by integrationTestScenario is invalid", errName, errKind), "integrationTestScenario.Name", integrationTestScenario.Name)
+					_, err = gitops.MarkSnapshotAsFailed(a.client, a.context, a.snapshot, fmt.Sprintf("Resource %s (%s) associated with snapshot %s is invalid: %s.", errName, errKind, a.snapshot.Name, err))
```

Comment:
We should inform user which ITS is invalid instead of snapshot, Status is recorded in snapshot, so user knows which snapshot is affected, but user doesn't know which ITS is it causing. Error log above provides more useful info than msg in snapshot IMHO.

Reply:
ack yes good point, changed in most recent patch.

Reply:
I don't see it